tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -387,11 +387,24 @@ func (rb *IptablesRuleBuilder) buildRestore(rules []Rule) string {
 	for _, r := range rules {
 		chainTable := fmt.Sprintf("%s:%s", r.chain, r.table)
 		// Create new chain if key: `chainTable` isn't present in map
-		if !chainTableLookupMap.Contains(chainTable) {
+		if !chainTableLookupMap.InsertContains(chainTable) {
 			// Ignore chain creation for built-in chains for iptables
-			if _, present := constants.BuiltInChainsMap[r.chain]; !present {
+			if !constants.BuiltInChainsMap.Contains(r.chain) {
 				tableRulesMap[r.table] = append(tableRulesMap[r.table], fmt.Sprintf("-N %s", r.chain))
-				chainTableLookupMap.Insert(chainTable)
+			}
+		}
+	}
+	// Also look for chains we jump to, even if we have no rules in them
```

Comment:
@bleggett @leosarra this is maybe dumb but seemed like the best option. WDYT?

I was worried about removing the `-J ISTIO_PRERT` since we could easily introduce bugs here if we add some rules to the chain (as a dev, I would probably not think to make sure the jump rule was added!), and I was worried our cleanup logic would not handle it well

Reply:
> I was worried about removing the -J ISTIO_PRERT since we could easily introduce bugs here if we add some rules to the chain

Might be worth splitting rule creation into ingress/egress and categorically skipping the former in this scenario. 

If we are actually not hooking ingress at all, we will never have rules in `PREROUTING` anyway.

Reply:
My concern was we maybe would... we do for DNS for example. Not in `nat`, though, so maybe moot

Reply:
Ah, I missed DNS is doing that. That's annoying.

I still think we should be fine if we just decompose this func better: `inbound/outbound/DNS` and that will go a ways towards de-footgunning.

> I was worried about removing the -J ISTIO_PRERT since we could easily introduce bugs here if we add some rules to the chain (as a dev, I would probably not think to make sure the jump rule was added!),

Our logic should probably be smart enough to dynamically not-create the jump if the jump dest is empty, but I'm fine leaving it in place for now, jumps to empty chains are harmless.